File: Data Structures & Algorithms/can-place-flowers/submission-1.java
class Solution {
    public boolean canPlaceFlowers(int[] fb, int n) {
        int ans = 0;
        for(int i = 0; i < fb.length; i++)
        {
            if(fb[i] == 0 && fb.length > 1)
            {
                if(i == 0 && fb[i + 1] == 0)
                {
                    ans+=1;
                    fb[i] = 1;
                }
                else if(i == fb.length - 1 && fb[i - 1] == 0)
                {
                    ans+=1;
                    fb[i] = 1;
                }
                else if(i > 0 && i < fb.length - 1 && fb[i - 1] == 0 && fb[i + 1] == 0)
                {
                    ans+=1;
                    fb[i] = 1;
                }
            }
            else if(fb[i] == 0) ans+=1;
        }
        return ans >= n;
    }
}

File: Data Structures & Algorithms/boats-to-save-people/submission-1.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& nums, int k) {
        sort(nums.begin(),nums.end());
        int ans = 0;
        int i = 0;
        int j = nums.size() - 1;
        int sum;
        while(i <= j)
        {
            if(i == j)
            {
                ans+=1;
                continue;
            }
            sum = nums[i] + nums[j];
            if(sum > k) j--;
            else
            {
                i++;
                j--;
            }
            ans+=1;
            
        }
        return ans;
    }
};